### PR TITLE
[ADP-1963] move TxCBOR to the Read hierarchy

### PR DESCRIPTION
**Pull request by paolino**
- [x] redefine `Read.Tx` as `EraValue`
- [x] redefine `TxBOR` as `EraValue` , just to carry the era of the tx
- [x] redefine TxCBOR codecs to `Tx` as `EraFun`s

### CODE REVIEW

File: lib/wallet/src/Cardano/Wallet/Read/Tx/CBOR.hs [hunk 1]
```diff
@@ -1,100 +1,123 @@
...
+    { byronFun = f
+    , shelleyFun = f
+    , allegraFun = f
+    , maryFun = f
+    , alonzoFun = f
+    , babbageFun = f
+    }
+    where
+        f :: ToCBOR(TxT era) => Tx era -> K BL.ByteString  era
+        f = K . toLazyByteString . toCBOR . unTx
```
Comment (HeinrichApfelmus) [inline lib/wallet/src/Cardano/Wallet/Read/Tx/CBOR.hs]:
```suggestion
  where
    f :: ToCBOR (TxT era) => Tx era -> K BL.ByteString  era
    f = K . toLazyByteString . toCBOR . unTx
```
(Formatting:
`where` 2 spaces looks nicer.

A space between constraint `ToCBOR` and its argument helps readability.
)

File: lib/wallet/test/unit/Cardano/Wallet/DB/Arbitrary.hs [hunk 2]
```diff
@@ -416,10 +420,11 @@ arbitraryChainLength = 10
...
 instance Arbitrary TxCBOR where
-    arbitrary = TxCBOR
-        <$> cbor
-        <*> (toEnum <$> choose (0,5))
-      where
+    arbitrary = do
+        c <- cbor
+        n <- elements knownEraIndices
+        pure $ fromRight' $ match eraValueSerialize (c,n)
+        where
```
Comment (HeinrichApfelmus) [inline lib/wallet/test/unit/Cardano/Wallet/DB/Arbitrary.hs]:
```suggestion
      where
```

(Formatting: `where` 2 spaces)

File: lib/wallet/src/Cardano/Wallet/Read/Tx/Certificates.hs [hunk 3]
```diff
@@ -1,68 +0,0 @@
...
-{-# LANGUAGE UndecidableInstances #-}
-
--- |
--- Copyright: © 2020-2022 IOHK
--- License: Apache-2.0
---
--- Raw certificate data extraction from 'Tx'
---
-
-module Cardano.Wallet.Read.Tx.Certificates
```
Comment (HeinrichApfelmus) [inline lib/wallet/src/Cardano/Wallet/Read/Tx/Certificates.hs]: Oh — was the deletion of this module here intentional? 🤔

File: lib/wallet/src/Cardano/Wallet/DB/Store/CBOR/Store.hs [hunk 4]
```diff
@@ -41,39 +59,45 @@ import Database.Persist
...
-        )
+type TxCBORRaw = (BL.ByteString, Int)
+
+i :: Iso' (BL.ByteString, Int) (ByteString, Word16)
+i = iso (toStrict *** fromIntegral) (fromStrict *** fromIntegral)
+
+toTxCBOR :: (TxId, TxCBOR) -> (CBOR, TxCBORRaw)
+toTxCBOR (id', tx) = let
+        r = build eraValueSerialize tx
+        in (uncurry (CBOR id') $ r ^. i, r)
```
Comment (HeinrichApfelmus) [inline lib/wallet/src/Cardano/Wallet/DB/Store/CBOR/Store.hs]:
```suggestion
toTxCBOR (id', tx) =
        let r = build eraValueSerialize tx
        in  (uncurry (CBOR id') $ r ^. i, r)
```

(Formatting: `let` in front of the line is a slightly easier to parse visually.)

**Review by HeinrichApfelmus — APPROVED** [review]
Looks good to me, thanks! 😊 Only minor formatting requests.

File: lib/wallet/src/Cardano/Wallet/Read/Tx/Certificates.hs (hunk 3 above)
Comment (paolino) [reply lib/wallet/src/Cardano/Wallet/Read/Tx/Certificates.hs]: no idea what it was doing there :shrug:

Comment (paolino) [reply lib/wallet/src/Cardano/Wallet/Read/Tx/Certificates.hs]: truth is that it was in good company, unneeded stuff for the future ... my bad. I put that back ..

**Comment by paolino** [comment]
bors try

CI failure:
```

   test\unit\Cardano\Wallet\Api\Server\TlsSpec.hs:118:13: 
  1) Cardano.Wallet.Api.Server.Tls, TLS Client Authentication, Deny client with wrong certificate if TLS is enabled
       predicate failed on expected exception: HttpException (HttpExceptionRequest Request {
         host                 = "127.0.0.1"
         port                 = 62658
         secure               = True
         requestHeaders       = []
         path                 = "/"
         queryString          = ""
         method               = "GET"
         proxy                = Nothing
         rawBody              = False
         redirectCount        = 10
         responseTimeout      = ResponseTimeoutDefault
         requestVersion       = HTTP/1.1
         proxySecureMode      = ProxySecureWithConnect
       }
        (InternalException Network.Socket.recvBuf: invalid argument (Invalid argument)))

  To rerun use: --match "/Cardano.Wallet.Api.Server.Tls/TLS Client Authentication/Deny client with wrong certificate if TLS is enabled/"

Randomized with seed 77324437
```

5 times ....

**Pull request merged**